internal/cli/internal/command/services/postgres/util.go

```diff
@@ -0,0 +1,46 @@
+package postgres
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"time"
+
+	"github.com/superfly/flyctl/internal/client"
+)
+
+func WaitForMachineState(ctx context.Context, client *client.Client, appId, machineId, state string) error {
+	fmt.Printf("Waiting for machine %q to reach a healthy state.\n", machineId)
+	timeout := time.After(2 * time.Minute)
+	tick := time.Tick(1 * time.Second)
+	for {
```

Comment:
I think a `for ctx.Err() == nil` loop in combination with `pause.For(ctx, 5*time.second)` would work better for you here, no? ([pause](https://github.com/azazeal/pause))

Reply:
It's unclear what the difference would be exactly.  Is there a clear benefit to using pause vs. ticks?

Reply:
`pause` would return once `ctx` is canceled or deadlined. `time.Sleep` would end up waiting for 5seconds when the user hits Ctrl+C (as an example).

Reply:
I'd still need a top-level timeout though, right?  Is there a better way to handle this?

```golang
ctx, cancel := context.WithTimeout(parent, time.Duration(time.Minute*2))
defer cancel()

go func(ctx context.Context) {
    for ctx.Err() == nil {
	  pause.For(ctx, time.Second)
  
	  machines, err := client.API().ListMachines(ctx, appID, "")
	  if err != nil {
		  logger.Debugf("failed retrieving machines: %v", err)
  
		  continue
	  }
  
	  for _, machine := range machines {
		  if machine.ID == machineID {
			  if machine.State == state {
				  cancel()
			  }
  
			  fmt.Fprintf(io.Out, "Machine state: %q, wanted: %q\n", machine.State, state)
		  }
	  }
    }
}(ctx)

select {
    case <-ctx.Done():
        if ctx.Err() == context.DeadlineExceeded {
          return ctx.Err()
        }

       return ctx.Err()
}
```